### ISSUE TITLE
Can't quite `flip` `always`

### ISSUE THREAD

**Issue by dinocarl** [ramda/ramda#3159]
I might expect that `flip(always)` would return the second argument given it, like so

```js
const inverseAlways = flip(always);
inverseAlways(1)(2); // 2
```

as a sort of delayed `identity`. And, in fact, `always(identity)` works exactly this way, and I would expect these 2 functions to be extensionally equivalent. However, the above version [returns `undefined`][]. It becomes more obvious why when we add a [second parameter][] to the first invocation:

```js
inverseAlways(1, 3)(2); // 3
```

It seems that the issue is that `always` uses a `_curry1` and a [hard-coded function][] as the return of the first argument. Could the implementation of the function be updated like so

```js
var always = _curry2(function always(a, b) {
  return a;
});
export default always;
```

Here is a version running in the [REPL][].

This would allow for both calling contexts of a flipped `always` to be used:

```js
const inverseAlways = flip(always);

inverseAlways(9, 10); // 10
inverseAlways(9)(10); // 10
```

where the second would remain equivalent to

```js
const inverseAlways = always(identity);

inverseAlways(9)(10); // 10
```

which would continue to need to be written this way.

  [hard-coded function]: https://github.com/ramda/ramda/blob/v0.27.1/source/always.js#L24
  [returns `undefined`]: https://ramdajs.com/repl/#?const%20inverseAlways%20%3D%20flip%28always%29%3B%0AinverseAlways%281%29%282%29
  [second parameter]: https://ramdajs.com/repl/#?const%20inverseAlways%20%3D%20flip%28always%29%3B%0AinverseAlways%281%2C%203%29%282%29
  [REPL]: https://ramdajs.com/repl/#?const%20alwys%20%3D%20curry%28%28a%2C%20b%29%20%3D%3E%20a%29%3B%0Aconst%20inverseAlways%20%3D%20flip%28alwys%29%3B%0A%0A%5B%0A%20%20alwys%289%2C%2010%29%2C%0A%20%20alwys%289%29%2810%29%2C%0A%20%20inverseAlways%289%2C%2010%29%2C%0A%20%20inverseAlways%289%29%2810%29%2C%0A%5D

**Comment by mattgrande** [comment]
I'm a bit confused about why you're trying to `flip` `always`. `always` is a unary function. `flip` expects a function with (at least) two arguments.

**Comment by dinocarl** [comment]
`always` does accept a second argument, it just doesn't care what that second one is and throws it away, and accepts it in hard-coded curried way. In other words, it's not unary, like `identity` is. A flipped version would do that same thing, but throw away the first.

**Comment by CrossEye** [comment]
I clearly see *what* you're doing.  I'm just really not sure why.  I think of `always` as equivalent to `const` in some other places -- accepting a value and returning a function that always returns that value, useful in some cases where you need to supply a function to a generic processor, but in this case you need it to return that one value, and slightly -- but only slightly -- more readable than `() => val`.

So while at first glance I don't have a major objection, I don't want to bother thinking too deeply about it unless you can articulate a reason.

**Comment by dinocarl** [comment]
Thanks for taking the time to respond @CrossEye.

The reasons are ultimately somewhat shallow and I'm not sure if that helps or hurts not making you think too deeply about the issue.

1. It seems to go against the idea ramda's support of multiple call syntaxes.
2. It's surprising that `always` can't have `flip` modify it, and (Haskell) `const` can.
3. A flipped `always` can have compatible semantics with `filter` and `map` (among others).
4. While `always(identity)` permits the above already, that formulation requires a bit more comprehension as to why and how it works, as opposed to the much more descriptive, `flip(always)`.

In short, this feels like a very minor issue to me that already has simple workarounds, but also has a simple fix that wouldn't invalidate those workarounds.

**Comment by CrossEye** [comment]
@dinocarl:

I would have concerns about regressions in doing this.  Assuming we rewrote this as

```js
const always = curry((a, b) => a)
```

then anyone calling 

```js
const foo = always (42);

// later

foo ()
```

would be in for a surprise, since this will return the equivalent of `foo` and not `42`.  The trouble is that JS is inherently variadic, and we would certainly expect to sometimes see `foo` used as a nullary function.

How could we write `always` to allow us to `flip` it as you would like but not break such existing code?

**Comment by dinocarl** [comment]
You're absolutely right - I missed this most essential use case. And I'm not sure it's possible to accommodate this. I'll try some variants, but so far I have not been able to crack this.

**Comment by dinocarl** [comment]
I think it makes sense to close this. I haven't found a reasonable way to do this outside of `always(identity)` (which we already had going into this). Thanks for your time and consideration, @CrossEye.
